### PR TITLE
Allow theme to be explicitly disabled

### PR DESCRIPTION
**Pull request by ntkme**
This PR allows theme to be explicitly disabled via `theme: null`.

When a source site contains its theme, files from the default theme `jekyll-theme-primer` will still ends up in the dest site. The most annoying part of this issue (pages-themes/primer#15) is that `assets/css/style.css` from `jekyll-theme-primer` would show up in `site.pages` collection (e.g. `sitemap.xml`). The upstream seems to have no interest in fixing that issue, and I don't really care either as long as I can avoid that buggy theme.

According to [Jekyll::Site.configure_theme](https://github.com/jekyll/jekyll/blob/29dc190fcb41f13dfe9764d1d539a34657d44788/lib/jekyll/site.rb#L435), supposedly a user should be able to disable any theme by having `theme: null` in `_config.yml`. However, due to the way configurations are merged in [GitHubPages::Configuration.effective_config](https://github.com/github/pages-gem/blob/938b95356d0dda654d2323ff7075b38b20ce597b/lib/github-pages/configuration.rb#L100) (eventually the logic in [Jekyll::Utils.deep_merge_hashes](https://github.com/jekyll/jekyll/blob/29dc190fcb41f13dfe9764d1d539a34657d44788/lib/jekyll/utils.rb#L319-L320)), an explicit `theme: null` will still be overrode by `theme: "jekyll-theme-primer"`.

Changing the behavior of `Jekyll::Utils.deep_merge_hashes` on explicit `null` in yaml (or `nil` in ruby) is unrealistic, as it is so fundamental that tons of existing code depend on its current behavior. Thus, the best solution is fixing it the after configuration merging in this repository.

#482 mentioned using a hack of `theme: false`, it works because [Jekyll::Site.configure_theme](https://github.com/jekyll/jekyll/blob/29dc190fcb41f13dfe9764d1d539a34657d44788/lib/jekyll/site.rb#L441-L443) treats a boolean `false` as an invalid value and then fallback to `nil`. This PR is intended to avoid the such hack and properly allow theme to be explicitly disabled.

### CODE REVIEW

**Comment by ntkme** [comment]
Test is failing on ruby 2.5 due to travis-ci/travis-ci#8969.

File: lib/github-pages/configuration.rb [hunk 1]
```diff
@@ -101,6 +101,11 @@ def effective_config(user_config)
           .fix_common_issues
           .add_default_collections
 
+        # Allow theme to be explicitly disabled via "theme: null"
+        if user_config.key?("theme") && user_config["theme"].nil?
+          config["theme"] = nil
```
Comment (parkr) [inline lib/github-pages/configuration.rb]: What do you think about something like:

```text
config["theme"] = user_config["theme"] if user_config.key?("theme")
```

**Comment by ntkme** [comment]
@parkr Your version is more readable and less confusing, so I've happily followed your advise.

**Comment by ntkme** [comment]
@parkr @benbalter Any update?

**Comment by MikeMcQuaid** [comment]
Thanks @ntkme!

**Pull request merged**